### PR TITLE
PWGHF: taskCorrelationD0Hadrons.cxx: Add a condition to fill Toward, Transverse, and Away regions.

### PR DESCRIPTION
**Pull request by 1481014945**
Hello,
I want to make some changes to the previous histograms. The previous histograms had too many dimensions, making the storage too large. Therefore, I want to split them into three parts for storage.

### CODE REVIEW

**Review by fgrosa — CHANGES_REQUESTED** [review]
Hi @1481014945, to avoid issues with the binning, I would suggest you to use `ConfigurableAxis`, so that you can modify the binning from the configuration directly without changing the code further. You can see an example here: https://github.com/AliceO2Group/O2Physics/blob/d1149bb65c0435346e902efa092e34cb4aad220f/PWGHF/Tasks/taskCharmHadImpactPar.cxx#L38.

File: PWGHF/HFC/Tasks/taskCorrelationD0Hadrons.cxx [hunk 1]
```diff
@@ -249,11 +252,28 @@ struct HfTaskCorrelationD0Hadrons {
...
-        registry.fill(HIST("hCorInfoWithCorrelationState"), deltaPhi, deltaEta, ptD, ptHadron, poolBin, massD, isAutoCorrelated, efficiencyWeight);
-      }
-      if (signalStatus == ParticleTypeData::D0barOnly || (signalStatus == ParticleTypeData::D0D0barBoth)) {
-        registry.fill(HIST("hCorInfoWithCorrelationState"), deltaPhi, deltaEta, ptD, ptHadron, poolBin, massDbar, isAutoCorrelated, efficiencyWeight);
+      if (isTowardTransverseAway) {
+        if (ptHadron < leadingParticlePtMin) {
+          continue;
+        }
+        if (signalStatus == ParticleTypeData::D0Only || (signalStatus == ParticleTypeData::D0D0barBoth)) {
+          if (-o2::constants::math::PI / 3. < deltaPhi && deltaPhi < o2::constants::math::PI / 3.) {
```
Comment (vkucera) [inline PWGHF/HFC/Tasks/taskCorrelationD0Hadrons.cxx:260]: Looks like you can just use `std::abs`.

File: PWGHF/HFC/Tasks/taskCorrelationD0Hadrons.cxx [hunk 2]
```diff
@@ -249,11 +252,28 @@ struct HfTaskCorrelationD0Hadrons {
...
+          }
+        }
+        if (signalStatus == ParticleTypeData::D0barOnly || (signalStatus == ParticleTypeData::D0D0barBoth)) {
+          if (-o2::constants::math::PI / 3. < deltaPhi && deltaPhi < o2::constants::math::PI / 3.) {
+            registry.fill(HIST("hToward"), massD, ptD, isAutoCorrelated, efficiencyWeight);
+          } else if (o2::constants::math::PI * 2. / 3. < deltaPhi && deltaPhi < o2::constants::math::PI * 4. / 3.) {
+            registry.fill(HIST("hAway"), massD, ptD, isAutoCorrelated, efficiencyWeight);
+          } else {
+            registry.fill(HIST("hTransverse"), massD, ptD, isAutoCorrelated, efficiencyWeight);
+          }
```
Comment (vkucera) [inline PWGHF/HFC/Tasks/taskCorrelationD0Hadrons.cxx:275]: If you plan to add more code branched based on the region, I would suggest to define a function, which takes `deltaPhi` as argument and returns the region, which can be an `enum` constant.

**Comment by 1481014945** [comment]
Thank you for your advice. I will try to do it.

**Review by deepathoms — CHANGES_REQUESTED** [review]
I have no comments for the PR

File: PWGHF/HFC/Tasks/taskCorrelationD0Hadrons.cxx [hunk 3]
```diff
@@ -249,11 +252,28 @@ struct HfTaskCorrelationD0Hadrons {
         continue;
       }
       //==============================================================================================================
-      if (signalStatus == ParticleTypeData::D0Only || (signalStatus == ParticleTypeData::D0D0barBoth)) {
-        registry.fill(HIST("hCorInfoWithCorrelationState"), deltaPhi, deltaEta, ptD, ptHadron, poolBin, massD, isAutoCorrelated, efficiencyWeight);
-      }
-      if (signalStatus == ParticleTypeData::D0barOnly || (signalStatus == ParticleTypeData::D0D0barBoth)) {
-        registry.fill(HIST("hCorInfoWithCorrelationState"), deltaPhi, deltaEta, ptD, ptHadron, poolBin, massDbar, isAutoCorrelated, efficiencyWeight);
+      if (isTowardTransverseAway) {
```
[the remark this answers is not in the document]
Comment (1481014945) [reply PWGHF/HFC/Tasks/taskCorrelationD0Hadrons.cxx:272]: This old part was added by me before. This time, the change adds an "if" statement to reduce the impact on the original code.

[the remark this answers is not in the document]
Comment (deepathoms) [reply PWGHF/HFC/Tasks/taskCorrelationD0Hadrons.cxx:272]: Please check with other authors of the code to make sure there are no conflicts with the D0-h correlations. I would suggest to have a different function for your analysis where you are looking for D0 in different regions w.r.t a trigger particle.

**Review by fgrosa — APPROVED** [review]

**Comment by vkucera** [comment]
> @vkucera Hello vkucera! I have tried to modify it, but I am not sure if it meets your suggestions.

Hi @1481014945 , it looks good to me. Thanks.

**Comment by fgrosa** [comment]
@deepathoms could you please check if it is ok for you as well? Thanks!

File: PWGHF/HFC/Tasks/taskCorrelationD0Hadrons.cxx (hunk 3 above)
[the remark this answers is not in the document]
Comment (samrangy) [reply PWGHF/HFC/Tasks/taskCorrelationD0Hadrons.cxx:272]: Looks fine for me.

**Review by deepathoms — APPROVED** [review]

**Pull request merged**